### PR TITLE
Tag support implementation for fetching and invalidate cache entries marked by tags

### PR DESCRIPTION
**Pull request by Smart-Core**
Very useful things.

### CODE REVIEW

File: lib/Doctrine/Common/Cache/Cache.php [hunk 1]
```diff
@@ -73,10 +73,11 @@ public function contains($id);
      * @param int    $lifeTime The lifetime in number of seconds for this cache entry.
      *                         If zero (the default), the entry never expires (although it may be deleted from the cache
      *                         to make place for other entries).
+     * @param array $tags
      *
      * @return bool TRUE if the entry was successfully stored in the cache, FALSE otherwise.
      */
-    public function save($id, $data, $lifeTime = 0);
+    public function save($id, $data, $lifeTime = 0, array $tags = []);
```
Comment (Ocramius) [inline lib/Doctrine/Common/Cache/Cache.php]: This is an interface change, and a BC break. Not mergeable, sorry.

**Comment by Koc** [comment]
@Smart-Core have you looked at Symfony 3.2 PSR6 implementation? Have you considered switching to it? Also Symfony provide more efficient method of invalidation by tag.

**Comment by Ocramius** [comment]
Looking at this again, I don't really see a need for it inside a generic lib like doctrine/cache.

This is a very specific use-case that has very very few consumers, generally. A namespaced cache key reduces the use-case scenario massively too.

Closing, therefore.

@Koc PSR-6 got downvoted by doctrine because of it's useless complexity, but doesn't include tagging AFAIK.

**Comment by Smart-Core** [comment]
it's a pity

**Comment by Ocramius** [comment]
@Smart-Core always consider the 80/20 here: supporting all use-cases doesn't bring much, whereas a custom cache implementation can support them with really not that much coding (see this PR) after all.

This is more like a 5% scenario, so this library is not a good fit for the feature.

**Pull request closed without merging**